x/logic/prolog/assert.go

```diff
@@ -154,15 +154,37 @@ func AssertList(term engine.Term, env *engine.Env) (engine.Term, error) {
 // AssertPair resolves a term as a pair and returns the pair components.
 // If conversion fails, the function returns nil and the error.
 func AssertPair(term engine.Term, env *engine.Env) (engine.Term, engine.Term, error) {
+	return assertTuple2WithFunctor(term, AtomPair, AtomTypePair, env)
+}
+
+// AssertKeyValue resolves a term as a key-value and returns its components, the key as an atom,
+// and the value as a term.
+// If conversion fails, the function returns nil and the error.
+func AssertKeyValue(term engine.Term, env *engine.Env) (engine.Atom, engine.Term, error) {
+	k, v, err := assertTuple2WithFunctor(term, AtomKeyValue, AtomKTypeKeyValue, env)
+	if err != nil {
+		return AtomEmpty, nil, err
+	}
+
+	key, err := AssertAtom(k, env)
+
+	return key, v, err
+}
```

Comment:
Hmm, okay. The code, as it stands, handles the error correctly since it returns the error from the `atom` assertion and returns the `empty` atom as the first returned value. However, since the value `v` is also being returned instead of `nil`, I agree that it would be better to correct it as suggested. This will likely require adding a specific test case for that.